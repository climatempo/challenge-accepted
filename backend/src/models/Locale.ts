import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity({ name: "locales" })
export class Locale {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column()
  identification!: number;

  @Column()
  name!: string;

  @Column()
  state!: string;

  @Column()
  latitude!: number;

  @Column()
  longitude!: number;
}
