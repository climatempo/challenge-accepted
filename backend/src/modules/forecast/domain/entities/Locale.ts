import { z } from 'zod';
import { Latitude } from '../valueObject/Latitude';
import { LocaleId } from '../valueObject/LocaleId';
import { LocaleName } from '../valueObject/LocaleName';
import { Longitude } from '../valueObject/Longitude';
import { State } from '../valueObject/State';

const localeSchema = z.object({
  id: z.instanceof(LocaleId),
  name: z.instanceof(LocaleName),
  state: z.instanceof(State),
  latitude: z.instanceof(Latitude).optional().nullable(),
  longitude: z.instanceof(Longitude).optional().nullable(),
});

type LocaleProps = z.infer<typeof localeSchema>;

export class Locale {
  private props: LocaleProps;

  constructor(props: LocaleProps) {
    this.props = {
      ...props,
    };

    localeSchema.parse(this);
  }

  get id(): LocaleId {
    return this.props.id;
  }
  get name(): LocaleName {
    return this.props.name;
  }
  get state(): State {
    return this.props.state;
  }
  get latitude(): Latitude | null {
    return this.props.latitude || null;
  }
  get longitude(): Longitude | null {
    return this.props.longitude || null;
  }

  public toPrimitives() {
    return {
      id: this.id.value,
      name: this.name.value,
      state: this.state.value,
      latitude: this.latitude?.value,
      longitude: this.longitude?.value,
    };
  }
}
