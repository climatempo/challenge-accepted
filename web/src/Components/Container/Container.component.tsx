import { useAppSelector } from '../../hooks';
import Empty from '../Empty';
import InstantWeather from '../InstantWeather';
import DailyWeathers from '../DailyWeathers';

function Container() {
    const locale = useAppSelector(state => state.locale.value);

    function renderContainer() {
        return <>
            <InstantWeather/>
            <DailyWeathers/>
        </>
    }

    function renderEmpty() {
        return <Empty/>
    }

    return <div className="mt-8 sm:mt-16 py-6 px-8 relative transition-all mx-auto max-w-[1024px] w-full gap-16">
        { locale ? renderContainer() : renderEmpty() }
    </div>;
}

export default Container;