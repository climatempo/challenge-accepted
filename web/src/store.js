import Vue from "vue";
import Vuex from "vuex";
import localeStore from "./store/locale";
import weatherStore from "./store/weather";

Vue.use(Vuex);

export default new Vuex.Store({
  state: {},
  mutations: {},
  actions: {},
  modules: [
      localeStore,
      weatherStore
  ]
});
