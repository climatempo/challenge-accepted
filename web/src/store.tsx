import { configureStore } from '@reduxjs/toolkit';
import themeReducer from './Reducers/Theme.reducer';
import temperatureReducer from './Reducers/Temperature.reducer';
import precipitationReducer from './Reducers/Precipitation.reducer';

const store = configureStore({
    reducer: {
        theme: themeReducer,
        temperature: temperatureReducer,
        precipitation: precipitationReducer,
    }
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export default store;