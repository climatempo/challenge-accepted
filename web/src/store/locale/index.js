import mutations from './mutations'
import actions from './actions'
import getters from './getters'

/**
 * Store de localizaçeõs
 *
 * @type mixins
 */
const localeStore = {
    state: {
        locales: []
    },
    mutations: mutations,
    actions: actions,
    getters: getters
};

export default localeStore
