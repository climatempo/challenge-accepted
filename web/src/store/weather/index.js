import mutations from './mutations'
import actions from './actions'
import getters from './getters'

/**
 * Store de previs√µes
 *
 * @type mixins
 */
const weatherStore = {
    state: {
        weather: []
    },
    mutations: mutations,
    actions: actions,
    getters: getters
};

export default weatherStore
