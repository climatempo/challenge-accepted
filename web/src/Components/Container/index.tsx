import Container from './Container.component';

export default Container;