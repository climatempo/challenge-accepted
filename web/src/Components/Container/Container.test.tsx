import renderer from 'react-test-renderer';
import './../../Tests/watchMedia.mock';
import Container from './index';
import { DetailedLocale, setLocale } from '../../Reducers/Locale.reducer';
import { Provider } from 'react-redux';
import store from '../../store';
import { SnackbarProvider } from 'notistack';

test("Container", () => {
    const dummyLocale = {
        "idlocale": 8600,
        "idcity": 3664,
        "capital": false,
        "idcountry": 7,
        "ac": "BR",
        "country": "Brasil",
        "uf": "MG",
        "city": "Ch√°cara",
        "region": "SE",
        "seaside": false,
        "latitude": -21.672,
        "longitude": -43.222,
        "tourist": false,
        "agricultural": false,
        "base": "cities",
        "searchPoints": 200
    } as DetailedLocale;

    const reduxStore = store;

    const container = renderer.create(
      <Provider store={reduxStore}>
          <SnackbarProvider maxSnack={4}>
              <Container/>
          </SnackbarProvider>
      </Provider>
    );

    expect(container.toJSON()).toMatchSnapshot();

    // Added dummy city info
    reduxStore.dispatch(setLocale(dummyLocale));
    const containerPopulated = renderer.create(
      <Provider store={reduxStore}>
          <SnackbarProvider maxSnack={4}>
            <Container/>
          </SnackbarProvider>
      </Provider>
    );

    expect(containerPopulated.toJSON()).toMatchSnapshot();
});
