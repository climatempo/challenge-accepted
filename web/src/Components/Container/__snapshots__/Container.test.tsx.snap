// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container 1`] = `
<div
  className="mt-8 sm:mt-16 py-6 px-8 relative transition-all mx-auto max-w-[1024px] w-full gap-16"
>
  <div
    className="flex flex-col justify-center items-center pt-8 pb-16 sm:py-20 text-center gap-8"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 hidden md:block css-1sra7t5-MuiTypography-root"
    >
      Clima Tempo Challenge
    </h2>
    <h6
      className="MuiTypography-root MuiTypography-h6 hidden md:block css-2ulfj5-MuiTypography-root"
    >
      Use a barra de pesquisa para achar sua cidade.
    </h6>
    <h4
      className="MuiTypography-root MuiTypography-h4 md:hidden css-5lbw0b-MuiTypography-root"
    >
      Clima Tempo
      <br />
      Challenge
    </h4>
    <h6
      className="MuiTypography-root MuiTypography-h6 md:hidden css-2ulfj5-MuiTypography-root"
    >
      Use a barra de pesquisa para achar sua cidade.
    </h6>
  </div>
</div>
`;

exports[`Container 2`] = `
<div
  className="mt-8 sm:mt-16 py-6 px-8 relative transition-all mx-auto max-w-[1024px] w-full gap-16"
>
  <div
    className="flex gap-10 items-center max-w-[720px] mx-auto flex-col lg:flex-row"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary mx-auto css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </div>
  <div
    className="mt-24 flex flex-col"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary mx-auto css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </div>
</div>
`;
